### PR TITLE
Portable sleeping bags

### LINKED ISSUE #26371: New item: Hiking Rucksack. Provides a way for the Camper to store his/her default stuff.
**Is your feature request related to a problem? Please describe.**  

I'm frustrated when I play as the Camper. They doesn't have enough room to store his starting inventory. On starting, the sleeping bag is worn. While you can take this off, it results in a "No room in your inventory for the sleeping bag. Drop it? (y/n)" which forces you to wield it. 

Since I am sure the bag is supposed to be in your inventory, the fact that the Camper cannot fit it in their inventory is a concern.

**Describe the solution you'd like**  

I seek to remedy that with a new piece of clothing: the hiking rucksack.

It holds 15.00L of items, between the Military Rucksack and the Backpack, and will have Encumbrance 12. Made of cotton and leather, it will have 4/4 bash/cut protection: slightly less than a leather backpack while being higher than a normal backpack. The volume is 3.00L, in between a normal backpack and the military rucksack.

Description:

>A large hiking backpack made of cotton and leather. Provides a fairly large amount of storage.

This will be given to the Camper and show up in houses, sporting goods shops and military surplus shops.

**Describe alternatives you've considered**  

I've considered giving the Camper a MOLLE Pack, but these and military rucksacks are reserved for the military.

**EDIT**

From [this](https://www.reddit.com/r/cataclysmdda/comments/9qmidh/camper_doesnt_have_enough_space_to_store_his_stuff/) Reddit thread:

 * A rucksack that has the ability to hold a sleeping bag/roll mat (in the same way that sheaths hold weapons, or a toolbelt holds tools)

* The sleeping bag should be activatable to roll it up and make it smaller with low cover, protection and durability and still be wearable. (needs knowledge of item_transform)

* 15 storage for 12 encumberance is too strong, survivor backpacks/rucksacks and such are 3/3 behind 6+ tailoring skill, I would say 17 encumberance for 15 storage sounds ok if you want it that sturdy as a 4/4

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
```SUMMARY: Content "Portable sleeping bags"```

#### Purpose of change
Resolves #26371

#### Describe the solution
Sleeping bags can be rolled up for transport.
They have a strap, so they can be worn, and carried around.
To use them to sleep you need to unroll them first.
Added automatic refresh of encumbrance and bodytemp when transforming worn items.

#### Describe alternatives you've considered
Hiker's backpack, but it'd be unbalanced so I went for the alternative way of doing that.

#### Additional context
N/A

### CODE REVIEW

**Comment by nexusmrsep** [comment]
Good call, will do that.

File: data/json/items/armor.json [hunk 1]
```diff
@@ -7758,7 +7758,32 @@
...
+      "type": "transform",
+      "target": "sleeping_bag_roll",
+      "msg": "You roll up the sleeping bag, preparing it for transport."
+    }
+  },
+  {
+    "id": "sleeping_bag_roll",
+    "type": "ARMOR",
+    "name": "rolled sleeping bag",
+    "description": "A large sleeping bag rolled up for transport. It has a strap to carry it with.",
```
Comment (ZhilkinSerg) [inline data/json/items/armor.json]:
```suggestion
    "description": "A large sleeping bag rolled up for transport.  It has a strap to carry it with.",
```

File: data/json/items/armor.json [hunk 2]
```diff
@@ -7778,7 +7803,37 @@
...
+      "type": "transform",
+      "target": "sleeping_bag_fur_roll",
+      "msg": "You roll up the fur sleeping bag, preparing it for transport."
+    }
+  },
+  {
+    "id": "sleeping_bag_fur_roll",
+    "type": "ARMOR",
+    "name": "rolled fur sleeping bag",
+    "description": "A large sleeping bag lined with fur, rolled for transport. It has a strap to carry it with.",
```
Comment (ZhilkinSerg) [inline data/json/items/armor.json]:
```suggestion
    "description": "A large sleeping bag lined with fur, rolled for transport.  It has a strap to carry it with.",
```

**Comment by Mecares** [comment]
They would benefit from using "looks like".

**Pull request merged**